include/rmm/mr/device/per_device_resource.hpp

```diff
@@ -247,5 +275,175 @@ inline device_memory_resource* set_current_device_resource(device_memory_resourc
 {
   return set_per_device_resource(rmm::get_current_cuda_device(), new_mr);
 }
+
+/**
+ * @brief Get the `device_async_resource_ref for the specified device.
+ *
+ * Returns a `device_async_resource_ref` for the specified device. The initial resource_ref
+ * references a `cuda_memory_resource`.
+ *
+ * `device_id.value()` must be in the range `[0, cudaGetDeviceCount())`, otherwise behavior is
+ * undefined.
+ *
+ * This function is thread-safe with respect to concurrent calls to `set_per_device_resource_ref`,
+ * `get_per_device_resource_ref`, `get_current_device_resource_ref`,
+ * `set_current_device_resource_ref` and `reset_current_device_resource_ref. Concurrent calls to any
```

Comment:
This one is copy-pasted and needs a few fixes.
```suggestion
 * `set_current_device_resource_ref` and `reset_current_device_resource_ref`. Concurrent calls to any
```